File: cpp/masterpiece_cpp/05-Exercise_Theory/05-Q10.cpp
/* 이론 문제
Q10
다음의 각 문제가 별도로 실행될 때 array 배열은 어떻게 되는가?
*/
#include <iostream>

using namespace std;

int array[] = {0, 2, 4, 6, 8, 10, 12, 14, 16, 18};
int& f(int n) {
    return array[n];
}

int main() {

//  (1)
    f(9) = 100;             // array[9] = 100

//  (2)
    for (int i = 1; i < 9; i++)
        f(i) = f(i) + 2;    // array[1] = 4
                            // array[2] = 6
                            // array[3] = 8
                            // array[4] = 10
                            // array[5] = 12
                            // array[6] = 14
                            // array[7] = 16
                            // array[8] = 18
//  (3)
    int v = f(0); v = 100;  // array[0] = 0

//  (4)
    f(f(2)) = 0;            // array[6] = 0

    return 0;
}

File: python/doit_JumpToPython/chap05/05-Q10.py
"""
Q10
os 모듈을 사용하여 다음과 같이 동작하도록 코드를 작성해 보자.

1. C:\doit 디렉터리로 이동한다.

2. dir 명령을 실행하고 그 결과를 변수에 담는다.

3. dir 명령의 결과를 출력한다.

"""
import platform
import os

if platform != 'windows':
    # os.chdir("~")
    f = os.popen("ls ~")
else:
    # os.chdir("c:\doit")
    f = os.popen("dir c:\")
print(f.read())
